pkg/utils/kube/scheme/typed.go

```diff
@@ -0,0 +1,72 @@
+package scheme
+
+import (
+	"reflect"
+	"sync"
+
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/util/managedfields"
+	"sigs.k8s.io/structured-merge-diff/v4/typed"
+)
+
+// ResolveParseableType will build and return a ParseableType object
+// based on the given gvk and GvkParser. If the given gvkParser is nil
+// it will return a DeducedParseableType which is not suitable for
+// calculating diffs. Will use the statically defined schema for k8s
+// built in types. Will rely on the given gvk parser for CRD schemas.
+func ResolveParseableType(gvk schema.GroupVersionKind, parser *managedfields.GvkParser) *typed.ParseableType {
+	if parser == nil {
+		return &typed.DeducedParseableType
+	}
+	pt := resolveFromStaticParser(gvk, parser)
+	if pt == nil {
+		return parser.Type(gvk)
+	}
+	return pt
+}
+
+func resolveFromStaticParser(gvk schema.GroupVersionKind, parser *managedfields.GvkParser) *typed.ParseableType {
+	gvkNameMap := getGvkMap(parser)
+	name := gvkNameMap[gvk]
+
+	p := StaticParser()
```

Comment:
It's definitely a micro-optimization, but could we avoid this if `name == ""` by returning nil?

Reply:
sorry but not sure I understand the suggestion. Returning `nil` from where?

Reply:
Ff name == nil on line 33, we return nil on line 34 and don’t use the output of StaticParser from line 32. We could check name before calling StaticParser and save some time/memory.